internal/core/bfs.go

```diff
@@ -8,88 +8,88 @@ import (
 	"github.com/microsoft/typescript-go/internal/collections"
 )
 
-type BreadthFirstSearchResult[N comparable] struct {
+type BreadthFirstSearchResult[N any] struct {
 	Stopped bool
 	Path    []N
 }
 
-type breadthFirstSearchJob[N comparable] struct {
+type breadthFirstSearchJob[N any] struct {
 	node   N
 	parent *breadthFirstSearchJob[N]
 }
 
-type BreadthFirstSearchLevel[N comparable] struct {
-	jobs *collections.OrderedMap[N, *breadthFirstSearchJob[N]]
+type BreadthFirstSearchLevel[K comparable, N interface{ Key() K }] struct {
```

Comment:
Two things:
1. I don't know if this is the best way to do this. It would also be possible to pass `Key` as a function, I think, or maybe some other setup.

2. I'm not sure if `Key` is the best name for this, so open to suggestions.

Reply:
I'll try that and see how it goes.